src/test/kotlin/cacophony/automata/minimalization/PartitionRefinementTest.kt

```diff
@@ -0,0 +1,186 @@
+package cacophony.automata.minimalization
+
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNotEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class PartitionRefinementTest {
+    private fun initRefinement(n: Int) = PartitionRefinement((0..<n).toList())
+
+    private val part =
+        object {
+            operator fun get(vararg values: Int): MutableSet<Set<Int>> = mutableSetOf(values.toSet())
+        }
+
+    private operator fun MutableSet<Set<Int>>.get(vararg values: Int): MutableSet<Set<Int>> =
+        apply {
+            this.add(values.toSet())
+        }
+
+    @Test
+    fun `getAllPartitions works after one refinement`() {
+        val pr = initRefinement(5)
+        pr.refine(listOf(0, 2, 3, 4))
+        assertEquals(part[0, 2, 3, 4][1], pr.getAllPartitions().toSet())
+    }
+
+    @Test
+    fun `getAllPartitions works after two refinements`() {
+        val pr = initRefinement(5)
+        pr.refine(listOf(0, 2, 3, 4))
+        pr.refine(listOf(3))
+        assertEquals(part[0, 2, 4][1][3], pr.getAllPartitions().toSet())
+    }
+
+    @Test
+    fun `getAllPartitions works after three refinements`() {
```

Comment:
You can do things like that in one test function. No need to create things like that